### PR TITLE
feat: pluggable per-backend structure via entry points

### PR DESCRIPTION
**Pull request by henryiii**
:robot: _AI text below_ :robot:

Closes #168.

## What

Refactor the per-build-backend logic out of two scattered `if/elif` chains into a pluggable **`backends/` package** — one module per backend — discovered via an **entry-point group**, so backends are extensible by users. Loosely modeled on `check-manifest`, but generalized across many build backends.

## Before

- `backends.py::backend_ignored_patterns()` — read each backend's exclude config, filtered the git-only set.
- `sdist.py::default_sdist_ignore()` — yielded sdist-only ignores (`*.dist-info`, setuptools `*.egg-info`/`setup.cfg`, autogenerated version files).

A single backend's knowledge was split across two files, and there was no extension point.

## After

```
src/check_sdist/backends/
├── __init__.py          # Backend, load_backends(), resolve_backend()
├── _base.py             # Backend Protocol, BaseBackend, glob_filter, pathspec_filter
├── setuptools.py  flit.py  hatchling.py  scikit_build_core.py
├── pdm.py  poetry.py  maturin.py
```

- **`Backend` protocol**: `build_backends` (for `auto` detection) + `git_only_excludes()` + `sdist_only_ignores()`. Each backend module owns *both* halves of its diff.
- **Entry points**: registered under `check_sdist.backends`, keyed by the build-backend string. Discovery is 3.9-compatible. Users register their own backend the same way; `build-backend` then accepts its name.
- **`compare()`** resolves the backend once up front (so an unknown explicit backend errors before the slow SDist build); `*.dist-info` stays backend-agnostic.

## Decisions (per discussion)

1. Entry-point keys = build-backend strings — **no change to existing `build-backend` config values**.
2. **Strict per-backend** version-file detection — only the resolved backend contributes its sdist-only ignores (a small correctness improvement over the old scan-all-backends behavior).
3. **Clean break** — removed `backend_ignored_patterns`/`default_sdist_ignore`; rewrote `tests/test_backends.py` (now also covers registry/resolution + relocated version-file tests) and removed the now-empty `tests/test_sdist.py`.

Schema `build-backend` relaxed to `anyOf: [enum, string]` so custom backend names validate while built-ins keep editor autocomplete (regenerated via `nox -s generate_schema`).

## Verification

- `uv run nox -s tests` → 57 passed
- `prek -a --quiet` → clean
- mypy → clean on all new modules
- `uv run check-sdist` on this repo → matches git

### CODE REVIEW

File: src/check_sdist/backends/__init__.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                return backend
+        return None
+
+    if selector in backends:
+        return backends[selector]
+    for backend in backends.values():
+        if selector in backend.build_backends:
+            return backend
+
+    msg = f"Unknown backend: {selector} - please add support in check_sdist.backends"
```
Comment (henryiii) [inline src/check_sdist/backends/__init__.py]:
```suggestion
    msg = f"Unknown backend: {selector} - request addition in check_sdist.backends or make a plugin"
```
